core/storage/ipld/cids_ipld.cpp

```diff
@@ -179,22 +183,59 @@ namespace fc::storage::ipld {
     row.key = key;
     row.offset = car_offset;
     row.max_size64 = maxSize64(item.size());
-    if (!common::write(writable, item)) {
-      spdlog::error("CidsIpld.put write error");
-      outcome::raise(ERROR_TEXT("CidsIpld.put: write error"));
-    }
-    if (!writable.flush().good()) {
-      spdlog::error("CidsIpld.put flush error");
-      outcome::raise(ERROR_TEXT("CidsIpld.put: flush error"));
-    }
     car_offset += item.size();
+    carPut(row, std::move(item));
     written.insert(row);
     if (flush_on != 0 && written.size() >= flush_on) {
       written_lock.unlock();
       asyncFlush();
     }
   }
 
+  void CidsIpld::carPut(const Row &row, Bytes &&item) {
+    std::unique_lock lock{car_flush_mutex};
+    car_queue.emplace(row.offset.value(), std::move(item));
+    if (car_queue.size() >= car_flush_on) {
+      carFlush(std::adopt_lock);
+    }
+  }
+
+  bool CidsIpld::carGet(const Row &row, Bytes &value) const {
+    std::shared_lock lock{car_flush_mutex};
+    const auto it{car_queue.find(row.offset.value())};
+    if (it == car_queue.end()) {
+      return false;
+    }
+    BytesIn item{it->second};
+    BytesIn input;
+    const CbCid *hash;
+    if (codec::uvarint::readBytes(input, item)
+        && codec::cbor::light_reader::readCborBlake(hash, input)) {
```

Comment:
I suggest to rework `readCborBlake()` from out parameter to return based approach.
`const CbCid *hash = readCborBlake(input)`
`if (hash != nullptr) { ... }`

Reply:
@Alexey-N-Chernyshov you have TODO for that in `readCborBlake()`
What do you think?

Reply:
Yes, I prefer not to use out parameters, and here the function like `bool isCborBlakeObject(input)` would be more meaningful to check if input has valid object.
But I think the point of discussion is out of PR scope.

Reply:
I agree that we could fix it in separate ticket